lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2173,38 +2178,37 @@ export class UserAgentApplication {
   // If pCacheStorage is separated from the class object, or passed as a fn param, scopesUtils.ts can be created
 
   /**
-   * Used to validate the scopes input parameter requested  by the developer.
+   * Used to validate the scopes input parameter requested  by the developer and
+   * to remove openid and profile from the list of scopes passed by the developer.
+   * These scopes are added by default.
+   * 
    * @param {Array<string>} scopes - Developer requested permissions. Not all scopes are guaranteed to be included in the access token returned.
    * @ignore
    * @hidden
    */
   // TODO: Check if this can be combined with filterScopes()
-  private validateInputScope(scopes: Array<string>): string {
-    // Check that scopes is not an empty array
-    if (!scopes || scopes.length < 1) {
-      return "Scopes cannot be passed as an empty array";
+  private validateAndFilterInputScope(scopes: Array<string>): Array<string> {
+    if (scopes == null) {
+      return null;
     }
 
     // Check that scopes is an array object
     if (!Array.isArray(scopes)) {
-      throw new Error("API does not accept non-array scopes");
+      throw ClientConfigurationError.createScopesNonArrayError(scopes);
```

Comment:
so its okay to pass in nothing, but not okay to pass in an empty array?

Reply:
Yeah this was a misunderstanding, I will fix this in a future commit.